src/kv/untyped_tx_view.h

```diff
@@ -12,12 +12,37 @@ namespace kv::untyped
   using SerialisedKeyHasher = std::hash<SerialisedEntry>;
 
   using VersionV = kv::VersionV<SerialisedEntry>;
-  using State =
-    kv::State<SerialisedEntry, SerialisedEntry, SerialisedKeyHasher>;
+
+  struct k_size
+  {
+    uint32_t operator()(const SerialisedEntry& key)
+    {
+      return sizeof(uint64_t) + key.size();
+    };
+  };
+
+  struct v_size
+  {
+    uint32_t operator()(const kv::untyped::VersionV& value)
+    {
+      return sizeof(uint64_t) + sizeof(value.version) + value.value.size();
+    };
+  };
+
+  using State = kv::State<
+    SerialisedEntry,
+    SerialisedEntry,
+    SerialisedKeyHasher,
+    k_size,
```

Comment:
It's unfortunate if we have to add these templates to all the kv types, and not just `Snapshot`. Why is this necessary?